### PR TITLE
drop support for 0.3

### PR DESCRIPTION
**Pull request by mlubin**
Don't see any need to hold this up. Note we require 0.4-rc2 or greater, so we don't have to worry about supporting prereleases.

### CODE REVIEW

**Comment by IainNZ** [comment]
Glorious

**Comment by joehuchette** [comment]
:+1:

**Comment by mlubin** [comment]
There are *a lot* of deprecation warnings in the solver wrappers.

File: src/norms.jl [hunk 1]
```diff
@@ -66,18 +66,7 @@ _build_norm{C,V}(P,terms::Vector{GenericAffExpr{C,V}}) = GenericNorm(P,terms)
...
-        _build_norm(Lp,AffExpr[])  # Punt
-    else
-        new_terms = Array(typeof(terms[1]), length(terms))
-        for i in 1:length(terms)
-            new_terms[i] = terms[i]
-        end
-        _build_norm(Lp,new_terms)
-    end
-end
+_build_norm(Lp, terms::Vector{GenericAffExpr}) = _build_norm(Lp, [terms...])
```
Comment (mlubin) [inline src/norms.jl]: @joehuchette, is this comment still true?

Comment (joehuchette) [inline src/norms.jl]: Pretty sure @IainNZ added this

Comment (IainNZ) [inline src/norms.jl]: Worth checking I guess

Comment (mlubin) [inline src/norms.jl]: How do we test?

**Comment by IainNZ** [comment]
Master:
```
idunning$ jl test/perf/norms.jl
  0.086521 seconds (1.55 M allocations: 211.580 MB, 23.59% gc time)
```

With long version commented out
```
dhcp-18-189-40-211:JuMP idunning$ jl test/perf/norms.jl
  0.079035 seconds (1.55 M allocations: 211.580 MB, 25.16% gc time)
```

**Comment by mlubin** [comment]
LGTM

**Pull request merged**